app/javascript/dashboard/components/ui/TimeAgo.vue

```diff
@@ -36,86 +30,55 @@ export default {
   },
   data() {
     return {
-      timeago: '',
+      timeAgo: '',
       timer: null,
     };
   },
   computed: {
-    currentDate() {
-      return new Date();
-    },
-    previousDate() {
-      return new Date(this.timestamp * 1000);
-    },
-
-    hasMinutesDiff() {
-      const minutes = differenceInMinutes(this.currentDate, this.previousDate);
-      return minutes < 60;
-    },
-    hasHourDiff() {
-      const hours = differenceInHours(this.currentDate, this.previousDate);
-      return hours >= 1 && hours <= 24;
-    },
-    hasDayDiff() {
-      const days = differenceInDays(this.currentDate, this.previousDate);
-      return days >= 1 && days <= 31;
-    },
-    hasMonthDiff() {
-      const months = differenceInCalendarMonths(
-        this.currentDate,
-        this.previousDate
-      );
-      return months >= 1 && months <= 6;
-    },
-    hasYearDiff() {
-      const years = differenceInCalendarYears(
-        this.currentDate,
-        this.previousDate
-      );
-      return years >= 1;
-    },
     refreshTime() {
-      if (this.refresh) {
-        if (this.hasMinutesDiff && !this.hasHourDiff) {
-          return MINUTE_IN_MSEC;
-        }
-        if (this.hasHourDiff && !this.hasDayDiff) {
-          return HOUR_IN_MSEC;
-        }
-        if (this.hasDayDiff && !this.hasMonthDiff) {
-          return DAY_IN_MSEC;
-        }
-        if (this.hasMonthDiff && !this.hasYearDiff) {
-          return MONTH_IN_MSEC;
-        }
-        if (this.hasYearDiff) {
-          return YEAR_IN_MSEC;
-        }
+      const timeDiff = differenceInMilliseconds(
+        new Date(),
+        new Date(this.timestamp * 1000)
+      );
+      if (
+        timeDiff > MINUTE_IN_MILLI_SECOND &&
+        timeDiff < HOUR_IN_MILLI_SECOND
+      ) {
+        return MINUTE_IN_MILLI_SECOND;
+      }
+      if (timeDiff > HOUR_IN_MILLI_SECOND && timeDiff < DAY_IN_MILLI_SECOND) {
+        return HOUR_IN_MILLI_SECOND;
+      }
+      if (timeDiff > DAY_IN_MILLI_SECOND && timeDiff < MONTH_IN_MILLI_SECOND) {
+        return DAY_IN_MILLI_SECOND;
+      }
+      if (timeDiff > MONTH_IN_MILLI_SECOND && timeDiff < YEAR_IN_MILLI_SECOND) {
+        return MONTH_IN_MILLI_SECOND;
+      }
+      if (timeDiff > YEAR_IN_MILLI_SECOND) {
+        return YEAR_IN_MILLI_SECOND;
       }
```

Comment:
Please write the comparison in reverse order.

time > year -> return year seconds
time > month -> return month seconds

This way, you wouldn't need to check for `time > hour && time < day.`

Reply:
Done https://github.com/chatwoot/chatwoot/pull/5253/commits/d9cf0b645453499a7d6cc0ea4cfee402f48f5420

Reply:
@iamsivin Do we need to set the time interval if the time difference is greater than month/year?
cc @pranavrajs

Reply:
@pranavrajs Do we need to do that?